### PR TITLE
Microsoft/users/aujensen/msbuild clean fix

### PR DESCRIPTION
**Pull request by austin-jensen**
Fixes the issue in #8948 so Msbuild and Vsbuild tasks don't get invoked twice when the clean flag is set

### CODE REVIEW

File: Tasks/Common/MSBuildHelpers/InvokeFunctions.ps1 [hunk 1]
```diff
@@ -19,30 +19,21 @@ function Invoke-BuildTools {
...
+            
+            $splat = @{ }
                 if ($CreateLogFile) {
-                    $splat["LogFile"] = "$file-clean.log"
+                    $splat["LogFile"] = "$file.log"
                 }
 
                 if ($LogFileVerbosity) {
                     $splat["LogFileVerbosity"] = $LogFileVerbosity
                 }
```
Comment (moswald) [inline Tasks/Common/MSBuildHelpers/InvokeFunctions.ps1]: This indentation is off now.

File: Tasks/Common/MSBuildHelpers/InvokeFunctions.ps1 [hunk 2]
```diff
@@ -42,7 +29,16 @@ function Invoke-BuildTools {
                 $splat["LogFileVerbosity"] = $LogFileVerbosity
             }
 
-            Invoke-MSBuild -ProjectFile $file -MSBuildPath $MSBuildLocation -AdditionalArguments $MSBuildArguments -NoTimelineLogger:$NoTimelineLogger @splat
+            if ($Clean) {
```
Comment (brcrista) [inline Tasks/Common/MSBuildHelpers/InvokeFunctions.ps1]: I think a comment on the approach would be helpful:

"If we need to clean, make sure we don't build any additional targets (/t:) twice"

Comment (brcrista) [reply Tasks/Common/MSBuildHelpers/InvokeFunctions.ps1]: With these long command lines, it's hard to see that they're actually the same thing.  I think a DRY approach is clearer:

```pwsh
if ($Clean) {
    Invoke-MSBuild -ProjectFile $file -Targets Clean -MSBuildPath $MSBuildLocation -AdditionalArguments $MSBuildArguments -NoTimelineLogger:$NoTimelineLogger @splat
}

# If we cleaned and passed /t targets, we don't need to run them again
if (!$Clean -or -not $MSBuildArguments.Contains("/t")) {
    Invoke-MSBuild -ProjectFile $file -MSBuildPath $MSBuildLocation -AdditionalArguments $MSBuildArguments -NoTimelineLogger:$NoTimelineLogger @splat
}
```

File: Tasks/Common/MSBuildHelpers/InvokeFunctions.ps1 [hunk 3]
```diff
@@ -42,7 +29,14 @@ function Invoke-BuildTools {
...
+                Invoke-MSBuild -ProjectFile $file -Targets Clean -MSBuildPath $MSBuildLocation -AdditionalArguments $MSBuildArguments -NoTimelineLogger:$NoTimelineLogger @splat
```
Comment (mperssonPTI) [inline Tasks/Common/MSBuildHelpers/InvokeFunctions.ps1]: Doesn't `-Targets Clean` overwrite the default behaviour? With the "Clean" option selected, do we actually still build? My code is no longer being compiled with this option selected so I have deselected it and added "/t:Clean;Build" to MSBuild arguments as a workaround.

**Pull request merged**